plugins/module_utils/cmci.py

```diff
@@ -408,6 +388,7 @@ def init_url(self):  # type: () -> str
     def init_request_params(self):  # type: () -> Optional[Dict[str, str]]
         return None
 
+    # TODO: why do we have this and/or parsing twice?
```

Comment:
Did you want to raise an issue for this and remove the TODO?

Reply:
I was actually gonna try and fix it, forgot I'd left the TODO in there... let me see if I can use the same stuff twice!